### PR TITLE
fix: Front Bug Fixes

### PR DESCRIPTION
**Pull request by midhun-aot**
- Handled Small Screen Login Button Style Issue

 - Hiding Location Details From Site Details If No Snapshot purchased

 - Adding Disabled Styles For Add To Cart & Add To Folio in Search Page When No Row Selected

 - Changing notification to top center
 
 - Disabling Tabs for not logged in user
 
 - Show location details to not logged in user 

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Backend](https://nr-site-registry-172-backend.apps.silver.devops.gov.bc.ca)
- [Frontend](https://nr-site-registry-172-frontend.apps.silver.devops.gov.bc.ca)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Backend](https://nr-site-registry-172-backend.apps.silver.devops.gov.bc.ca)
- [Frontend](https://nr-site-registry-172-frontend.apps.silver.devops.gov.bc.ca)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/app/components/navigation/Header.css [hunk 1]
```diff
@@ -142,3 +142,22 @@ header .other {
 ::-webkit-scrollbar-thumb {
   background: #888; /* color of the thumb */
 }
+
```
Comment (nupurdixit13) [inline frontend/src/app/components/navigation/Header.css:145]: Can we pls get rid of these extra lines?

File: frontend/src/app/components/navigation/Header.css [hunk 2]
```diff
@@ -142,3 +142,22 @@ header .other {
...
+
+
+@media (max-width: 576px) {
+
+  .customNavbarSM
+  {
+    display: flex;
+    flex-direction: column;
+
```
Comment (nupurdixit13) [inline frontend/src/app/components/navigation/Header.css]: Pls remove extra line

File: frontend/src/app/components/navigation/Header.css [hunk 3]
```diff
@@ -142,3 +142,22 @@ header .other {
...
+@media (max-width: 576px) {
+
+  .customNavbarSM
+  {
+    display: flex;
+    flex-direction: column;
+     
+  }
+
+  .header-right-corner-section {
```
Comment (nupurdixit13) [inline frontend/src/app/components/navigation/Header.css:158]: Is this being used anywhere?

File: frontend/src/app/features/details/SummaryForm.tsx [hunk 4]
```diff
@@ -23,7 +26,25 @@ const SummaryForm: FC<SummaryFormProps> = ({
   srMode,
   changeHandler,
 }) => {
-  const formRows: IFormField[][] = [
+
+  const userPurchasedSnapshot = useSelector(isUserPurchasedSnapshot);
+
+  const [finalFormConfiguration, SetFinalFormConfiguration] = useState<IFormField[][]>([]);
```
Comment (nupurdixit13) [inline frontend/src/app/features/details/SummaryForm.tsx]: pls follow camelCase convention for `SetFinalFormConfiguration`

File: frontend/src/app/features/details/SummaryForm.tsx [hunk 5]
```diff
@@ -1,11 +1,14 @@
...
+import React, { FC, useEffect, useState } from 'react';
 import {
   FormFieldType,
   IFormField,
 } from '../../components/input-controls/IFormField';
 import Form from '../../components/form/Form';
 import { Sites } from '../site/dto/Site';
 import './SummaryForm.css';
+import { useSelector } from 'react-redux';
+import { isUserPurchasedSnapshot } from './snapshot/SnapshotSlice';
```
Comment (nupurdixit13) [inline frontend/src/app/features/details/SummaryForm.tsx]: pls change `isUserPurchasedSnapshot` to `hasUserPurchasedSnapshot`

File: frontend/src/app/components/navigation/Header.css [hunk 6]
```diff
@@ -142,3 +142,22 @@ header .other {
...
   background: #888; /* color of the thumb */
 }
+
+
+
+
+
+@media (max-width: 576px) {
+
+  .customNavbarSM
```
Comment (nupurdixit13) [inline frontend/src/app/components/navigation/Header.css]: Can we pls give a meaningful name as SM at the end is not very intuitive?

File: frontend/src/app/features/site/Search.css [hunk 7]
```diff
@@ -265,3 +265,10 @@ div.search-parent {
   padding: var(--layout-padding-medium) var(--layout-padding-small)
     var(--layout-padding-medium) var(--layout-padding-small);
 }
+
+
+.disabled-btn
+{
+  color: var(--surface-color-primary-disabled, #EDEBE9);
```
Comment (nupurdixit13) [inline frontend/src/app/features/site/Search.css]: I don think this variable is defined in design token. Can you pls check at your end and use a suitable one?

File: frontend/src/app/components/navigation/Header.css (hunk 2 above)
Comment (midhun-aot) [reply frontend/src/app/components/navigation/Header.css]: Done

File: frontend/src/app/components/navigation/Header.css (hunk 3 above)
Comment (midhun-aot) [reply frontend/src/app/components/navigation/Header.css:158]: Yes

File: frontend/src/app/features/details/SummaryForm.tsx (hunk 4 above)
Comment (midhun-aot) [reply frontend/src/app/features/details/SummaryForm.tsx]: done

File: frontend/src/app/components/navigation/Header.css (hunk 6 above)
Comment (midhun-aot) [reply frontend/src/app/components/navigation/Header.css]: done

File: frontend/src/app/features/site/Search.css (hunk 7 above)
Comment (midhun-aot) [reply frontend/src/app/features/site/Search.css]: updated with right one.

File: frontend/src/app/features/details/SummaryForm.tsx (hunk 5 above)
Comment (midhun-aot) [reply frontend/src/app/features/details/SummaryForm.tsx]: Updated.

**Review by jaspalsingh-aot — APPROVED** [review]
LGTM

File: frontend/src/app/components/navigation/Header.tsx [hunk 8]
```diff
@@ -42,7 +42,7 @@ const Header = () => {
   };
 
   return (
-    <header className="navbar position-sticky ">
+    <header className="navbar position-sticky navbar-small-device">
```
Comment (nupurdixit13) [inline frontend/src/app/components/navigation/Header.tsx:45]: pls change this to  `navbar-for-small-device` as thats the name in css file

File: frontend/src/app/features/details/SummaryForm.tsx [hunk 9]
```diff
@@ -207,6 +203,33 @@ const SummaryForm: FC<SummaryFormProps> = ({
...
+      type: FormFieldType.Text,
+      label: 'Location Description',
+      placeholder: 'Location Description',
+      graphQLPropertyName: 'generalDescription',
+      value: '',
+      customLabelCss: 'custom-summary-lbl-text',
+      customInputTextCss: 'custom-summary-input-text',
+    }
+  ];
+
```
Comment (nupurdixit13) [inline frontend/src/app/features/details/SummaryForm.tsx:219]: Pls get rid of these extra lines

File: frontend/src/app/features/details/summary/Summary.tsx [hunk 10]
```diff
@@ -35,27 +35,28 @@ import {
...
 import SummaryInfo from './SummaryInfo';
-import { isUserPurchasedSnapshot } from '../snapshot/SnapshotSlice';
+import { hasUserPurchasedSnapshot } from '../snapshot/SnapshotSlice';
+import { UserType } from '../../../helpers/requests/userType';
 
 const Summary = () => {
   const auth = useAuth();
 
-  const isUserPurchasedSite = useSelector(isUserPurchasedSnapshot);
+  const isUserPurchasedSite = useSelector(hasUserPurchasedSnapshot);
```
Comment (nupurdixit13) [inline frontend/src/app/features/details/summary/Summary.tsx:52]: Pls change this to `hasUserPurchasedSite`

**Pull request merged**